navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -158,6 +159,54 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
         legJson.put("distance", Helper.round(distance, 1));
     }
 
+
+    /**
+     * fix the first IntersectionDetail.
+     *
+     * The first Intersection of the first step should only have 1 bearing and one "out" entry
+     *
+     */
+    private static List<PathDetail> fixFirstIntersectionDetail( List<PathDetail> intersectionDetails) {
+        List<PathDetail> list = new ArrayList<>();
+
+        // job1: find out the interesting part of the intersectionDetails
+        if (intersectionDetails.size() < 2) {
+            // LOG Error? User didnt ask for intersections ?
```

Comment:
This case could maybe happen if you exactly hit the end of a one way street. Unlikely but possible? So I am not sure if this is an error case, we could log it on info if we want to look into this?